integration/spark-common/src/main/scala/org/apache/spark/CarbonInputMetrics.scala

```diff
@@ -35,20 +35,28 @@ class CarbonInputMetrics extends InitInputMetrics{
     var inputMetrics: InputMetrics = _
     // bytes read before compute by other map rdds in lineage
     var existingBytesRead: Long = _
+    var recordCount: Long = _
```

Comment:
Better takes as AtomicLong

Reply:
CarbonInputMetrics doesn't need syncronization, only inputMetrics (spark) inside that need synchronization